src/core/operations/ECDSAVerify.mjs

```diff
@@ -0,0 +1,120 @@
+/**
+ * @author cplussharp
+ * @copyright Crown Copyright 2021
+ * @license Apache-2.0
+ */
+
+import Operation from "../Operation.mjs";
+import OperationError from "../errors/OperationError.mjs";
+import r from "jsrsasign";
+
+/**
+ * ECDSA Verify operation
+ */
+class ECDSAVerify extends Operation {
+
+    /**
+     * ECDSAVerify constructor
+     */
+    constructor() {
+        super();
+
+        this.name = "ECDSA Verify";
+        this.module = "Ciphers";
+        this.description = "Verify a message against a signature and a public PEM encoded EC key.";
+        this.infoURL = "https://wikipedia.org/wiki/Elliptic_Curve_Digital_Signature_Algorithm";
+        this.inputType = "string";
+        this.outputType = "string";
+        this.args = [
+            {
+                name: "Input Format",
+                type: "option",
+                value: [
+                    "Auto",
+                    "ASN.1 HEX",
+                    "Concat HEX",
+                    "JSON"
+                ]
+            },
+            {
+                name: "Message Digest Algorithm",
+                type: "option",
+                value: [
+                    "SHA-256",
+                    "SHA-384",
+                    "SHA-512",
+                    "SHA-1",
+                    "MD5"
+                ]
+            },
+            {
+                name: "ECDSA Public Key (PEM)",
+                type: "text",
+                value: "-----BEGIN PUBLIC KEY-----"
+            },
+            {
+                name: "Message",
+                type: "text",
+                value: ""
+            }
+        ];
+    }
+
+    /**
+     * @param {string} input
+     * @param {Object[]} args
+     * @returns {string}
+     */
+    run(input, args) {
+        let inputFormat = args[0];
+        const [, mdAlgo, keyPem, msg] = args;
+
+        if (keyPem.replace("-----BEGIN PUBLIC KEY-----", "").length === 0) {
+            throw new OperationError("Please enter a public key.");
+        }
+
+        // detect input format
+        if (inputFormat === "Auto") {
+            if (input.substr(0, 2) === "30" && r.ASN1HEX.isASN1HEX(input)) {
+                inputFormat = "ASN.1 HEX";
+            } else if (input.indexOf("{") !== -1) {
```

Comment:
It's probably better to write a function that just calls `JSON.parse()`, then wrap it in `try {} catch{}` to verify whether it's valid JSON or not.

Reply:
done